units/patch_intent.go

```diff
@@ -349,27 +351,101 @@ func (j *patchIntentProcessor) buildCliPatchDoc(ctx context.Context, patchDoc *p
 			patchDoc.Githash, projectRef.Identifier)
 	}
 
-	var reader io.ReadCloser
-	reader, err = db.GetGridFile(patch.GridFSPrefix, patchDoc.Patches[0].PatchSet.PatchFileId)
+	reader, err := db.GetGridFile(patch.GridFSPrefix, patchDoc.Patches[0].PatchSet.PatchFileId)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error getting patch file")
 	}
-	defer reader.Close()
-	bytes, err := ioutil.ReadAll(reader)
+	patchDoc.Patches[0].ModuleName = ""
+
+	// TODO: refactor this when migration is done. Don't run this if evergreen.CommitQueueAlias is true
+	patchBytes, err := ioutil.ReadAll(reader)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error parsing patch")
 	}
-
-	summaries, err := thirdparty.GetPatchSummaries(string(bytes))
+	summaries, err := thirdparty.GetPatchSummaries(string(patchBytes))
 	if err != nil {
 		return err
 	}
-	patchDoc.Patches[0].ModuleName = ""
 	patchDoc.Patches[0].PatchSet.Summary = summaries
 
+	if patchDoc.Alias == evergreen.CommitQueueAlias {
+		// refresh the reader
+		reader, err = db.GetGridFile(patch.GridFSPrefix, patchDoc.Patches[0].PatchSet.PatchFileId)
+		if err != nil {
+			return errors.Wrap(err, "error getting patch file")
+		}
+		defer reader.Close()
+		summaries, err := GetPatchSummariesByCommit(reader)
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":      "error getting summaries by commit",
+				"patch_id":     patchDoc.Id.Hex(),
+				"patch_doc_id": patchDoc.Patches[0].PatchSet.PatchFileId,
+			}))
+			patchDoc.Patches[0].PatchSet.Summary = summaries
```

Comment:
don't we want to do this when `err == nil`?

Reply:
👍